certbot-dns-godaddy/certbot_dns_godaddy/dns_godaddy.py

```diff
@@ -0,0 +1,168 @@
+        """
+        Delete a TXT record using the supplied information.
+
+        Note that both the record's name and content are used to ensure that similar records
+        created concurrently (e.g., due to concurrent invocations of this plugin) are not deleted.
+
+        Failures are logged, but not raised.
+
+        :param str domain_name: The domain to use to associate the record with.
+        :param str record_name: The record name (typically beginning with '_acme-challenge.').
+        :param str record_content: The record content (typically the challenge validation).
+        """
+
+        try:
+            domain = self._find_domain(domain_name)
+        except godaddypy.client.BadResponse as e:
+            logger.debug('Error finding domain using the Godaddy API: %s', e)
+            return
+
+        try:
+            domain_records = self.client.get_records(domain, record_type='TXT')
+
+            matching_records = [record for record in domain_records
+                                if record['type'] == 'TXT'
+                                and record['name'] == self._compute_record_name(domain, record_name)
+                                and record['data'] == record_content]
+        except godaddypy.client.BadResponse as e:
+            logger.debug('Error getting DNS records using the Godaddy API: %s', e)
+            return
+
+        for record in matching_records:
+            try:
+                logger.debug('Removing TXT record with name: %s', record['name'])
+                self.client.delete_records(domain, name=record['name'], record_type='TXT')
+            except godaddypy.client.BadResponse as e:
+                logger.warn('Error deleting TXT record %s using the Godaddy API: %s',
+                            record['name'], e)
+
+    def _find_domain(self, domain_name):
+        """
+        Find the domain object for a given domain name.
+
+        :param str domain_name: The domain name for which to find the corresponding Domain.
+        :returns: The Domain, if found.
+        :rtype: str
+        :raises certbot.errors.PluginError: if no matching Domain is found.
+        """
+
+        domain_name_guesses = dns_common.base_domain_name_guesses(domain_name)
+
+        domains = self.client.get_domains()
+
+        for guess in domain_name_guesses:
+            matches = [domain for domain in domains if domain == guess]
+
+            if len(matches) > 0:
+                domain = matches[0]
+                logger.debug('Found base domain for %s using name %s', domain_name, guess)
+                return domain
+
+        raise errors.PluginError('Unable to determine base domain for {0} using names: {1}.'
+                                 .format(domain_name, domain_name_guesses))
+
+    @staticmethod
+    def _compute_record_name(domain, full_record_name):
+        # The domain, from Godaddy's point of view, is automatically appended.
+        return full_record_name.rpartition("." + domain)[0]
```

Comment:
Now that two plugins need this logic, perhaps it would be good to move it to `dns_common` and avoid duplication.